contracts/0.8.9/sanity_checks/OracleReportSanityChecker.sol

```diff
@@ -559,16 +629,101 @@ contract OracleReportSanityChecker is AccessControlEnumerable {
         }
     }
 
-    function _checkOneOffCLBalanceDecrease(
+    function _addReportData(uint256 _timestamp, uint256 _exitedValidatorsCount, uint256 _negativeCLRebase) internal {
+        reportData.push(ReportData(
+            SafeCast.toUint64(_timestamp),
+            SafeCast.toUint64(_exitedValidatorsCount),
+            SafeCast.toUint128(_negativeCLRebase)
+        ));
+    }
+
+    function _sumNegativeRebasesNotOlderThan(uint256 _timestamp) internal view returns (uint256) {
+        uint256 sum;
+        for (int256 index = int256(reportData.length) - 1; index >= 0; index--) {
+            if (reportData[uint256(index)].timestamp > SafeCast.toUint64(_timestamp)) {
+                sum += reportData[uint256(index)].negativeCLRebaseWei;
+            } else {
+                break;
+            }
+        }
+        return sum;
+    }
+
+    function _exitedValidatorsAtTimestamp(uint256 _timestamp) internal view returns (uint256) {
+        for (int256 index = int256(reportData.length) - 1; index >= 0; index--) {
+            if (reportData[uint256(index)].timestamp <= SafeCast.toUint64(_timestamp)) {
+                return reportData[uint256(index)].totalExitedValidators;
+            }
+        }
```

Comment:
```suggestion
        for (uint256 index = reportData.length - 1; ; index--) {
            if (reportData[index].timestamp <= SafeCast.toUint64(_timestamp)) {
                return reportData[index].totalExitedValidators;
            }
            if (index == 0) {
                break;
            }
        }
```

Reply:
I agree with the comments, but we have this exact code already deployed in mainnet, and we need to maintain consistency between the code in the master branch and the verified deployed bytecode. So suggest moving these changes to a separate PR in develop branch